apps/vocabulary/main.go

```diff
@@ -0,0 +1,173 @@
+// Copyright 2020 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"flag"
+	"io/ioutil"
+	"log"
+	"os"
+
+	"github.com/golang/protobuf/proto"
+
+	pb "github.com/googleapis/gnostic/openapiv2"
+)
+
+func readDocumentFromFileWithName(filename string) (*pb.Document, error) {
+	data, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, err
+	}
+	document := &pb.Document{}
+	err = proto.Unmarshal(data, document)
+	if err != nil {
+		return nil, err
+	}
+	return document, nil
+}
+func addToMap(word string, mapName map[string]int) {
+	mapName[word] += 1
+}
+
+func processDocument(document *pb.Document, schemas map[string]int, operationId map[string]int, names map[string]int, properties map[string]int) {
+	if document.Definitions != nil {
+		for _, pair := range document.Definitions.AdditionalProperties {
+			addToMap(pair.Name, schemas)
+			processSchema(pair.Value, properties)
+		}
+	}
+	for _, pair := range document.Paths.Path {
+		v := pair.Value
+		if v.Get != nil {
+			processOperation(v.Get, operationId, names)
+		}
+		if v.Post != nil {
+			processOperation(v.Post, operationId, names)
+		}
+		if v.Put != nil {
+			processOperation(v.Put, operationId, names)
+		}
+		if v.Patch != nil {
+			processOperation(v.Patch, operationId, names)
+		}
+		if v.Delete != nil {
+			processOperation(v.Delete, operationId, names)
+		}
+	}
+}
+
+func processOperation(operation *pb.Operation, operationId map[string]int, names map[string]int) {
+	if operation.OperationId != "" {
+		addToMap(operation.OperationId, operationId)
+	}
+	for _, item := range operation.Parameters {
```

Comment:
consider breaking out the nested blocks into separate methods

Reply:
SGTM. Following existing patterns, this might moving lines 76-93 below into a "processParameter" function. 

Also, I think we should take Noah's suggestion and replace addToMap with direct updates to the map, e.g. `names[t2.BodyParameterName] += 1`